testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/CliControllerConfigTestCase.java

```diff
@@ -0,0 +1,441 @@
+ *
+ * @author Martin Schvarcbacher
+ */
+@RunWith(WildflyTestRunner.class)
+public class CliControllerConfigTestCase {
+
+    private static final Protocol TEST_RUNNER_PROTOCOL = HTTP_REMOTING;
+    private static final String CONTROLLER_ALIAS_NAME = "Test_Suite_Server1_Name";
+    private static final int INVALID_PORT = TestSuiteEnvironment.getServerPort() - 1;
+    private static final String DISCONNECTED_PROMPT = "[disconnected /]";
+    private static final String CONNECTED_PROMPT = "@";
+    private static final String READ_SERVER_STATE = ":read-attribute(name=server-state)";
+    private File tempJbossConfigFile;
+
+    @Before
+    public void prepareFiles() {
+        try {
+            tempJbossConfigFile = File.createTempFile("tmp-jboss-cli", ".xml");
+        } catch (IOException e) {
+            fail(e.getLocalizedMessage());
+        }
+    }
+
+    @After
+    public void deleteFiles() {
+        assertTrue("failed to delete test config file", tempJbossConfigFile.delete());
+    }
+
+    /**
+     * Default controller from jboss-cli.xml should be invalid to ensure settings are loaded from controller alias
+     */
+    @Test
+    public void testInvalidDefaultConfiguration() {
+        JbossCliConfig jbossCliConfig = new JbossCliConfig()
+                .setDefaultProtocol(REMOTE)
+                .setUseLegacyOverride(true)
+                .setDefaultControllerPort(INVALID_PORT)
+                .setDefaultControllerProtocol(HTTP_REMOTING);
+        jbossCliConfig.writeJbossCliConfig(tempJbossConfigFile);
+
+        CliProcessWrapper cli = getTestCliProcessWrapper(false);
+        try {
+            cli.executeInteractive();
+            boolean returnValue = cli.pushLineAndWaitForResults("connect", DISCONNECTED_PROMPT);
+            assertTrue(returnValue);
+        } catch (IOException ex) {
+            fail(ex.getLocalizedMessage());
+        } finally {
+            cli.destroyProcess();
+        }
+    }
+
+    /**
+     * Tests connection to a controller aliased in jboss-cli.xml using --controller,
+     * with all options (protocol, hostname, port) specified
+     */
+    @Test
+    public void testConnectToAliasedController() {
+        JbossCliConfig jbossCliConfig = new JbossCliConfig()
+                .setUseLegacyOverride(true)
+                .setDefaultProtocol(REMOTE)
+                .setDefaultControllerPort(INVALID_PORT)
+                .setDefaultControllerProtocol(HTTPS_REMOTING)
+                .setAliasControllerPort(TestSuiteEnvironment.getServerPort())
+                .setAliasControllerProtocol(TEST_RUNNER_PROTOCOL);
+        jbossCliConfig.writeJbossCliConfig(tempJbossConfigFile);
+
+        CliProcessWrapper cli = getTestCliProcessWrapper(true);
+        try {
+            cli.executeInteractive();
+            cli.pushLineAndWaitForResults("connect");
+            boolean returnState = cli.pushLineAndWaitForResults(READ_SERVER_STATE, CONNECTED_PROMPT);
```

Comment:
I'm using CONNECTED_PROMPT instead of null; for example, setting TestSuiteEnvironment.getServerPort() + 100 with pushLineAndWaitForResults(..,null) passes the test when it should fail.